offline/cable_cru_TRENDY.F90

```diff
@@ -1,1318 +1,1101 @@
+    CO2Year = MAX(LBOUND(CRU%CO2Vals, 1), CO2Year)
+    CO2Year = MIN(UBOUND(CRU%CO2Vals, 1), CO2Year)
+
+    CO2Air(:) = CRU%CO2Vals(CO2Year)
+  ELSE
+    ! The user has specified the year
+    READ(CRU%CO2Method, '(I)') CO2Year
+    CO2Air(:) = CRU%CO2Vals(CO2Year)
+  END IF
+END SUBROUTINE get_cru_co2
+
+!------------------------------------------------------------------------------!
+
+SUBROUTINE get_cru_ndep(CRU)
+  !*## Purpose
+  !
+  ! Read the NDep value from a specific year.
+  !
+  !## Method
+  !
+  ! Get the NDep values using the ```SPATIO_TEMPORAL_DATASET``` associated with
+  ! NDep and fill in the NDep data array.
+
+  ! Get the nitrogen deposition values from the MET data.
+  TYPE(CRU_TYPE), INTENT(INOUT)             :: CRU
+
+  ! The nitrogen deposition is a spatio_temporal_dataset netCDF dataset.
+  ! The data is unfortunately arranged, in that the frequency is yearly, but the
+  ! time index is monthly with month fractions i.e. 5.922..., 17.922... which
+  ! makes it very inconvenient to index. For now, we have to rely on the prior
+  ! knowledge that the dataset begins at 1850.
+  INTEGER         :: TimeIndex
+
+  ! Temporary array to store the full set of spatial data
+  REAL, DIMENSION(:, :), ALLOCATABLE        :: TmpArray
+
+  ! Iterator variable
+  INTEGER  :: GridCell
+
+  ! Status checker
+  INTEGER  :: ok
+
+  ! Which method are we using for Nitrogen deposition?
+  IF (CRU%NDepMethod == "Yearly") THEN
+    ! We're using the simulation year for the deposition data
+    TimeIndex = CRU%cYear - 1849
+  ELSE
+    ! We're using a specific year of data
+    READ(CRU%NDepMethod, '(I4)') TimeIndex
+    TimeIndex = TimeIndex - 1849
+  END IF
+
+  ! If the year is before the time period of the data, get the earliest
+  ! year.
+  TimeIndex = MAX(1, TimeIndex)
+
+  ! And finally grab the data. Since we're using landmask, we have to extract
+  ! the full set of data first and then read the unmasked points into the NDep
+  ! values.
+  IF (CRU%DirectRead) THEN
+
+    ApplyLandmaskDirect: DO GridCell = 1, CRU%mLand
+      ok = NF90_GET_VAR(CRU%NDepFID, CRU%NDepVID, CRU%NDepVals(GridCell),&
+      START = (/land_x(GridCell), land_y(GridCell), TimeIndex/))
+      IF (ok /= NF90_NOERR) THEN
+        CALL handle_err(ok, "Reading from NDep")
+      END IF
+    END DO ApplyLandMaskDirect
+  ELSE
+
+    ALLOCATE(TmpArray(CRU%xDimSize, CRU%yDimSize))
+
+    ok = NF90_GET_VAR(CRU%NDepFID, CRU%NDepVID, TmpArray,&
+      START = (/1, 1, TimeIndex/), COUNT = (/CRU%xDimSize, CRU%yDimSize, 1/))
+    IF (ok /= NF90_NOERR) THEN
+      CALL handle_err(ok, "Reading from NDep")
+    END IF
+
+    ApplyLandmaskIndirect: DO GridCell = 1, CRU%mLand
+      CRU%NDepVals(GridCell) = TmpArray(land_x(GridCell), land_y(GridCell))
+    END DO ApplyLandmaskIndirect
+  END IF
+
+
+END SUBROUTINE get_cru_ndep
+
+!!------------------------------------------------------------------------------!
+
+ELEMENTAL FUNCTION Esatf(TC)
```

Comment:
I'll open an issue to remove this function later. It already exists in the science code. I think there are several versions of it for various input type/dimensions. So no need to have this in here.

Reply:
Issue created #318

Reply:
This will be addressed in a later PR.